### ISSUE TITLE
Verification of PDF accessibility with Acrobat versions

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/bipm-si-brochure#152]
A BIPM user enquired whether PDF accessibility is compatible with "Adobe Acrobat XI Pro".

> I was able to read (each line separately) up until [this screenshot]

![image](https://user-images.githubusercontent.com/11865/132304212-51767d8f-d4a5-4ad2-8eb4-de488edd3d9c.png)

> but then clicking on the number (area) gave the following error:

<img width="715" alt="Screenshot 2021-09-07 at 3 39 52 PM" src="https://user-images.githubusercontent.com/11865/132304454-d74c2dfa-97cc-40f2-9a99-e3e9ec7dc4ca.png">

The user was able to read the file without problem after a system restart.

This issue is to investigate whether this problem is a compatibility problem with Acrobat XI Pro.

**Comment by Intelligent2013** [comment]
> but then clicking on the number (area) gave the following error:

> <img width="715" alt="Screenshot 2021-09-07 at 3 39 52 PM" src="https://user-images.githubusercontent.com/11865/132304454-d74c2dfa-97cc-40f2-9a99-e3e9ec7dc4ca.png">

> The user was able to read the file without problem after a system restart.

BIPM PDF contains attached files with .mml extension. Windows OS doesn't have the file association (by default) for .mml file extension. You need to set default application (Notepad for example) to open files with .mml extension:
- open the specified folder (`C:\....\Local\Temp\A9.tmp\`) in the Explorer, 
or 
- find any another .mml file, or create an empty file by the command `notepad sample.mml`
then
- right-click on .mml file and choose the default program to open .mml file (see how to: https://www.online-tech-tips.com/windows-10/how-to-change-file-associations-in-windows-10/)
- may be you need to re-run Acrobat (actually don't need to system reboot)

**Comment by ronaldtse** [comment]
Maybe we should not use `.mml` since it is not the expected MathML file extension.
<img width="541" alt="Screenshot 2021-09-07 at 6 28 15 PM" src="https://user-images.githubusercontent.com/11865/132329874-e553b8f8-50a2-42fc-bbe2-46096eca8f56.png">

We should probably use `.xml` instead. The proper IANA media type is `mathml+xml`:
https://www.iana.org/assignments/media-types/application/mathml+xml

**Comment by ronaldtse** [comment]
IANA item:

```
Type name
application

Subtype name
mathml+xml

Required parameters
None

Optional parameters
Same as charset parameter of application/xml as specified in [RFC3023]

Encoding considerations
The encoding considerations of application/xml as specified in [RFC3023] apply.

Security considerations
As with other XML types and as noted in [RFC3023] section 10, repeated expansion of maliciously constructed XML entities can be used to consume large amounts of memory, which may cause XML processors in constrained environments to fail.

...

Additional information

Magic number(s): see [RFC3023]

File extension(s):
.mml

Windows Clipboard Name:
MathML

Macintosh file type code(s)
MML
```

**Comment by ronaldtse** [comment]
@Intelligent2013 can you verify that if you have registered the `mml` extension on Windows, the MathML file can open? Thanks.

**Comment by ronaldtse** [comment]
Then we can close this.

**Comment by Intelligent2013** [comment]
Regarding the issue `I was able to read (each line separately) up until [this screenshot]`, I've made a few experiments to investigate this issue:
1. Created a simple Word document ([test.docx](https://github.com/metanorma/bipm-si-brochure/files/7120730/test.docx))
![image](https://user-images.githubusercontent.com/5559926/132328680-3756f1db-cca4-411c-99dc-1d42cffb7908.png)
with a few used fonts:
- main font is 'Calibri'
- `word3Arial` typed by Arial font
- `word4Times` typed by Times New Roman font
- `word5Stix` typed by STIX Two Math font
and converted it in the PDF by 'Save as Adobe PDF' (with using of Acrobat PDFMaker (install with Acrobat Pro), not File->Save As-> pdf!) - [Test_Adobe.pdf](https://github.com/metanorma/bipm-si-brochure/files/7120745/Test_Adobe.pdf)

'Read Out Load' feature 'mark' all whole line and read it fully:
![image](https://user-images.githubusercontent.com/5559926/132329318-a570c9f1-7de9-49ed-b1ad-8814770f1e2c.png)

2. Create simple xsl-fo file with the same sequence of words and fonts:
```xml
<fo:block font-family="Calibri">Word1 word2 <fo:inline font-family="Arial">word3Arial</fo:inline> <fo:inline font-family="Times New Roman">word4Times</fo:inline> <fo:inline font-family="STIX Two Math">word5Stix</fo:inline> word6 word7 word8</fo:block>
```
and PDF by Apache FOP - [test_FOP.pdf](https://github.com/metanorma/bipm-si-brochure/files/7120760/test_FOP.pdf)

'Read Out Load' feature 'mark' splits the string by 'fo:inline' element:
![image](https://user-images.githubusercontent.com/5559926/132329760-48579fa5-f909-459b-8128-0651fe6b42de.png)
![image](https://user-images.githubusercontent.com/5559926/132329796-3bf1352b-31a5-4bcc-82ff-e8d80f59870f.png)
![image](https://user-images.githubusercontent.com/5559926/132329822-19952f08-20e0-4715-9ea7-7f816ad9ee96.png)
![image](https://user-images.githubusercontent.com/5559926/132329856-8da98014-07a1-4d8d-9201-da4759ebcfa9.png)
![image](https://user-images.githubusercontent.com/5559926/132329882-00e880c0-545a-4e05-b3eb-cd5e6cd5d37c.png)


There is a difference between two PDF in tags structure:
1. PDF generated by Adobe:
![image](https://user-images.githubusercontent.com/5559926/132330419-cb4e3376-c969-4abc-b4c0-204e81875b2d.png)
2. PDF generated by Apache FOP:
![image](https://user-images.githubusercontent.com/5559926/132330580-cc1512f1-c363-4c2d-b9cc-362079dd71fe.png)

Need further investigation...

**Comment by Intelligent2013** [comment]
> 
> 
> @Intelligent2013 can you verify that if you have registered the `mml` extension on Windows, the MathML file can open? Thanks.

verified, file opens.

**Comment by ronaldtse** [comment]
Thanks @Intelligent2013 ! Closing.

**Comment by ronaldtse** [comment]
Whoops you're still checking the speech read along. Sorry.

**Comment by Intelligent2013** [comment]
Found some discussions. Looks like Acrobat has poor Read Out Loud support.
`Read This Page Only` and `Read to End of Document` features work ok, but individual lines ('by click', as mentioned in the initial post) isn't.

https://community.adobe.com/t5/acrobat-discussions/acrobat-pro-dc-reading-out-loud-stops-at-end-of-line-hyphens/m-p/11150243
```
We are aware of this issue and it will be fixed in the coming releases. For now, you can simply read the document via View -> Read Out Loud -> Clicking Read this page only OR Read to End of Document option. This will read the document continuously without interruption.
```


https://community.adobe.com/t5/acrobat-reader-discussions/read-out-loud-is-working-but-it-is-pausing-between-each-line/m-p/10182251
```
Acrobat's Read Aloud is not a screen reader.
Repeat: it is NOT a screen reader.

It does not access and render the content (aka, voice it) per current PDF/UA-1 and WCAG 2.0 accessibility standards. It does not have the majority of user controls that real screen readers have. And it consistently mis-reads basic content.

Do not waste your time testing a document for screen reader accessibility with Acrobat's Read Aloud tool. It will give you false positives and false negatives.

Many of us in the professional accessibility world have asked Adobe to either fix Read Aloud and make it a full-featured, compliant tool, or remove it from Acrobat entirely.

Do your testing with a real screen reader:

    JAWS by Freedom Scientific https://www.freedomscientific.com/Products/Blindness/JAWS
    NVDA (which is free) https://www.nvaccess.org/
```

**Comment by ronaldtse** [comment]
Closing this for now. Thanks @Intelligent2013 !
